common/formatting/token_partition_tree.h

```diff
@@ -82,6 +84,71 @@ std::ostream& operator<<(std::ostream& stream,
 // occurred, else nullptr.
 TokenPartitionTree* MoveLastLeafIntoPreviousSibling(TokenPartitionTree*);
 
+//
+// TokenPartitionTreeWrapper wrapper for ReshapeFittingSubpartitions
+//
+class TokenPartitionTreeWrapper {
+ public:
+  TokenPartitionTreeWrapper(const TokenPartitionTree& node_) : node(&node_) {}
+
+  // Grouping node
+  TokenPartitionTreeWrapper(const UnwrappedLine& uwline_) : node(nullptr) {
+    unwrapped_line = std::make_shared<UnwrappedLine>(uwline_);
+  }
+
+  TokenPartitionTreeWrapper() = delete;
+
+  // Helper accessor
+  const UnwrappedLine& Value() const {
+    if (node) {
+      return node->Value();
+    } else {
+      return *unwrapped_line;
+    }
+  }
+
+  // Try to append other node
+  UnwrappedLine Value(const TokenPartitionTree& other) const {
+    CHECK((node == nullptr) && (unwrapped_line != nullptr));
+    UnwrappedLine uw = *unwrapped_line;
+    uw.SpanUpToToken(other.Value().TokensRange().end());
+    return uw;
+  }
+
+  void Update(const VectorTree<TokenPartitionTreeWrapper>* child) {
+    const auto& token_partition_tree_wrapper = child->Value();
+    const auto& uwline = token_partition_tree_wrapper.Value();
+
+    unwrapped_line->SpanUpToToken(uwline.TokensRange().end());
+  }
+
+  void SetIndentationSpaces(int indent) {
+    CHECK((node == nullptr) && (unwrapped_line != nullptr));
+    unwrapped_line->SetIndentationSpaces(indent);
+  }
+
+  const TokenPartitionTree* Node() const {
+    return node;
+  }
+
+ private:
+  const TokenPartitionTree* node;
+  std::shared_ptr<UnwrappedLine> unwrapped_line;
+};
+
+// Core part of kAppendFittingSubPartitions algorithm
+// Append fitting subpartitions in simple greedy manner
+bool AppendFittingSubpartitions(
+    VectorTree<TokenPartitionTreeWrapper>& fitted_partitions,
+    const TokenPartitionTree& unfitted_partitions,
+    const verible::BasicFormatStyle& style,
+    bool wrap_second_line);
```

Comment:
I would move this function and the `TokenPartitionTreeUnwrapper` class into only the .cc file as implementation details, exposing only `ReshapeFittingSubpartitions` for direct use and direct testing.

If you really want to test through the internal function, consider moving these declarations into a `-internal.h` header (and have the `cc_library` and `cc_test` both depend on that in `srcs`).  That said, I still want to encourage testing through the public interfaces where possible.

Reply:
Direct testing is enough. But thanks for info about `-internal.h`